portal/src/OnboardingSurveyScreen.tsx

```diff
@@ -0,0 +1,491 @@
+import React, {
+  ComponentType,
+  useContext,
+  useState,
+  useMemo,
+  useCallback,
+} from "react";
+import { Routes, Navigate, Route, useNavigate } from "react-router-dom";
+import { useTheme, Label } from "@fluentui/react";
+import PrimaryButton from "./PrimaryButton";
+import DefaultButton, { DefaultButtonProps } from "./DefaultButton";
+import { FormattedMessage, Context } from "@oursky/react-messageformat";
+import FormTextField from "./FormTextField";
+import PhoneTextField, { PhoneTextFieldValues } from "./PhoneTextField";
+import { FormProvider } from "./form";
+import SurveyLayout from "./OnboardingSurveyLayout";
+import styles from "./OnboardingSurveyScreen.module.css";
+
+function ChoiceButton(props: DefaultButtonProps) {
+  const theme = useTheme();
+  const { checked } = props;
+  const styles = useMemo(() => {
+    return {
+      root: {
+        border: "none",
+        "--tw-ring-color": theme.semanticColors.variantBorder,
+      },
+      rootChecked: {
+        "--tw-ring-color": theme.palette.themePrimary,
+        color: theme.palette.themePrimary,
+        backgroundColor: theme.semanticColors.buttonBackground,
+      },
+      rootCheckedHovered: {
+        color: theme.palette.themePrimary,
+      },
+    };
+  }, [theme]);
+  return (
+    <DefaultButton
+      {...props}
+      toggle={true}
+      styles={styles}
+      className={checked ? "ring-2" : "ring-1"}
+    />
+  );
+}
+/*
+function CompoundChoiceButton (props: ButtonProps) {
+  const { buttonKey, pressed, onClick } = props;
+  const { renderToString } = useContext(Context);
+  return (
+    <CompoundButton
+      text={renderToString(buttonKey)}
+      secondaryText={renderToString(buttonKey + ".secondaryText")}
+      toggle
+      checked={pressed}
+      className={styles.compoundChoiceButton}
+      onClick={onClick}
+    />
+  )
+}
+*/
+
+interface ChoiceButtonGroupProps {
+  prefix: string;
+  choices: readonly string[];
+  state: Record<string, boolean>;
+  setChoice: (choice: Record<string, boolean>) => void;
+  processChoice: (
+    oldChoice: Record<string, boolean>,
+    selectedChoice: Readonly<string>
+  ) => Record<string, boolean>;
+  Button: ComponentType<DefaultButtonProps>;
+  isIndividualComponent: boolean;
```

Comment:
Instead of this prop, we separate ChoiceGroupVariantLabel, ChoiceGroupVariantQuestion, ChoiceGroupVariantMutipleChoice